### PR TITLE
feat: lang detection

### LINKED ISSUE #743: support language detection
**Actual Behaviour**

When users visit apisix.apache.org, our website will serve content in English by default.

**Expected Behaviour**

For users in Chinese, we could provide default Chinese contents for a quick and better involving experiment.

**Solution**

According to https://docusaurus.io/docs/i18n/introduction#i18n-non-goals, Docusaurus will not do language detection, see https://github.com/facebook/docusaurus/issues/5838#issuecomment-955278743 .

Maybe we could use this package? https://www.npmjs.com/package/detect-browser-language

### PR DESCRIPTION
**Pull request by SkyeYoung**
Fixes: #743 
Changes:

pls check https://github.com/apache/apisix-website/issues/743#issuecomment-1075880270

I didn't add a re-enable button, because after thinking, I find the step of detecting the language automatically should only happen when I first entered the website.

Screenshots of the change:

none, pls check preview.

Logic of Language Redirect:
Untitled-2022-02-18-2055

### CODE REVIEW

File: website/lang-redirect.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import ExecutionEnvironment from "@docusaurus/ExecutionEnvironment";
```
Comment (bzp2010) [inline website/lang-redirect.js]: Can we put it in the src folder? For example in a hook.

Comment (SkyeYoung) [reply website/lang-redirect.js]: Official seems to recommend the current method more.

Comment (SkyeYoung) [reply website/lang-redirect.js]: I‘ve moved the script to the clientModules directory under the src.

**Review by juzhiyuan — COMMENTED** [review]
Hi, @SkyeYoung. This PR has lots of style changes. 😅 How about this?

1. You could set up the prettier configuration and format all codes in one PR.
2. Submit this lang detection PR without extra codes style changes.

**Comment by SkyeYoung** [comment]
> Hi, @SkyeYoung. This PR has lots of style changes. 😅 How about this?
> 
> 1. You could set up the prettier configuration and format all codes in one PR.
> 2. Submit this lang detection PR without extra codes style changes.

Ok, I'll rollback it. I'm adding lint tools. :D

**Comment by bzp2010** [comment]
I found a strange problem. My broswer is in Chinese mode. Page will be redirect to the `//` and show 404 not found.

msedge_2022-03-25_16-24-45

**Comment by SkyeYoung** [comment]
> I found a strange problem. My broswer is in Chinese mode. Page will be redirect to the `//` and show 404 not found.

I have tried to solve this problem. Pls have a look.

**Review by bzp2010 — DISMISSED** [review]

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**